dlp/deid_test.py

```diff
@@ -0,0 +1,168 @@
+    harmful_string = 'My SSN is 372819127'
+
+    deid.deidentify_with_fpe(parent, harmful_string, alphabet='NUMERIC',
+                             wrapped_key=wrapped_key, key_name=key_name)
+
+    out, _ = capsys.readouterr()
+    assert 'My SSN is' in out
+    assert '372819127' not in out
+
+
+def test_deidentify_with_fpe_uses_surrogate_info_types(capsys):
+    harmful_string = 'My SSN is 372819127'
+
+    deid.deidentify_with_fpe(parent, harmful_string, alphabet='NUMERIC',
+                             wrapped_key=wrapped_key, key_name=key_name,
+                             surrogate_type=surrogate_type)
+
+    out, _ = capsys.readouterr()
+    assert 'My SSN is SSN_TOKEN' in out
+    assert '372819127' not in out
+
+
+def test_deidentify_with_fpe_ignores_insensitive_data(capsys):
+    harmless_string = 'My favorite color is blue'
+
+    deid.deidentify_with_fpe(parent, harmless_string, alphabet='NUMERIC',
+                             wrapped_key=wrapped_key, key_name=key_name)
+
+    out, _ = capsys.readouterr()
+    assert harmless_string in out
+# [END Deidentify with FPE]
+
+
+# [START Deidentify with date shift]
+def test_deidentify_with_date_shift(capsys):
+    deid.deidentify_with_date_shift(parent, input_csv_file=csv_file,
+                                    output_csv_file=output_csv_file,
+                                    lower_bound_days=date_shifted_amount,
+                                    upper_bound_days=date_shifted_amount,
+                                    date_fields=date_fields)
+
+    out, _ = capsys.readouterr()
+
+    assert 'Successful' in out
+    # read in csv??
+
+
+def test_deidentify_with_date_shift_using_context_field(capsys):
+    deid.deidentify_with_date_shift(parent, input_csv_file=csv_file,
+                                    output_csv_file=output_csv_file,
+                                    lower_bound_days=date_shifted_amount,
+                                    upper_bound_days=date_shifted_amount,
+                                    date_fields=date_fields,
+                                    context_field_id=csv_context_field,
+                                    wrapped_key=wrapped_key,
+                                    key_name=key_name)
+
+    out, _ = capsys.readouterr()
+
+    assert 'Successful' in out
+
+
+def test_deidentify_with_date_shift_requires_all_fields():
+    with pytest.raises(StandardError):
+        deid.deidentify_with_date_shift(parent, input_csv_file=csv_file,
+                                        output_csv_file=output_csv_file,
+                                        lower_bound_days=date_shifted_amount,
+                                        upper_bound_days=date_shifted_amount,
+                                        date_fields=date_fields,
+                                        context_field_id=csv_context_field,
+                                        key_name=key_name)
+# [END Deidentify with date shift]
```

Comment:
Remove all the START and END tags in the test file; they're only needed in the real sample file. They're used to mark regions that are auto-included in our documentation.

Reply:
done